File: src/python/triple.py
"""Contains the triple class which describes a relation between two words."""

class Triple:
    """The triple which describes a relation between two words."""

    def __init__(self, from_word : str, edge_name : str, to_word : str, score: float):
        self.fromWord = from_word
        self.edgeName = edge_name
        self.toWord = to_word
        self.score = score

File: src/ConceptMaps.DataModel/Triple.cs
namespace ConceptMaps.DataModel;

/// <summary>
/// Defines a triple which contains two words and their connection (<see cref="EdgeName"/>)
/// to each other as input for a directed graph.
/// </summary>
/// <remarks>
/// Data model between Python service and .NET application.
/// </remarks>
public record struct Triple(string FromWord, string? EdgeName, string ToWord, double Score = double.NaN);